src/Debug/MailerCollector.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Mailer\Debug;
+
+use Yiisoft\Mailer\MessageInterface;
+use Yiisoft\Yii\Debug\Collector\CollectorTrait;
+use Yiisoft\Yii\Debug\Collector\SummaryCollectorInterface;
+
+final class MailerCollector implements SummaryCollectorInterface
+{
+    use CollectorTrait;
+
+    private array $messages = [];
+
+    public function collectMessage(
+        MessageInterface $message,
+    ): void {
+        $this->messages[] = $message;
+    }
+
+    public function collectMessages(
+        array $messages,
+    ): void {
+        $this->messages = array_merge($this->messages, $messages);
+    }
+
+    public function getCollected(): array
+    {
+        if (!$this->isActive()) {
+            return [];
+        }
+        return [
+            'messages' => array_map(fn(MessageInterface $message) => [
+                'from' => (array) $message->getFrom(),
+                'to' => (array) $message->getTo(),
+                'subject' => $message->getSubject(),
+                'textBody' => $message->getTextBody(),
+                'htmlBody' => $message->getCharset() === 'quoted-printable'
+                    ? quoted_printable_decode($message->getHtmlBody())
+                    : $message->getHtmlBody(),
```

Comment:
Message keeps html body in raw format without encoding. Seems, not need make decode here.

Reply:
It's not true

Reply:
Why?